src/NexusMods.Paths/TemporaryFileManager.cs

```diff
@@ -136,8 +136,19 @@ public ValueTask DisposeAsync()
 
     private void Dispose_Impl()
     {
-        if (_deleteOnDispose && _fileSystem.FileExists(Path))
+        if (!_deleteOnDispose)
+            return;
+
+        if (_fileSystem.FileExists(Path))
```

Comment:
APIs like FileExists and DirectoryExists are a bit of an anti pattern. 

In the OS, they essentially do 90% of the work of basically opening a new file handle (navigating FileSystem trees, etc.) to then discard that work. So you can do that work again.

Just do `DeleteFile` and `DeleteDirectory` directly without checking for existence, and discard any exceptions you get. Achieves the same effect with less work.

-----------

As a fun fact, I remember Jared Parsons (the C# compiler guy) talk about this many, many years ago. I didn't understand it back then, but knowing what goes on under the hood, I understand a bit better now.

Reply:
Yeah, in this specific scenario, the common big OSes (all of the ones .NET runs on) always ship a different API for directories and files. Deleting a dir from file API will fail, and vice versa. So there shouldn't be any unexpected effects or a 'mixup' either.

Reply:
Skipping this since it becomes difficult to isolate issues with the path not existing (which is fine since there is nothing to cleanup) from issues of the actual deletion failing (file in use, permission denied).